src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -523,12 +524,11 @@ def update(self):
         # Flag that we have now checked for name clashes so that if there's
         # more than one PSyData node we don't fall over on the symbols
         # we've previous inserted.
-        self.root.psy_data_name_clashes_checked = True
+        routine_schedule.psy_data_name_clashes_checked = True
```

Comment:
pylint:
W:527, 8: Attribute 'psy_data_name_clashes_checked' defined outside __init__ (attribute-defined-outside-init)

Reply:
I've disabled the warning since this attribute exists only in a NemoSchedule and it is only the NEMO API that uses the `update` method. Once #435 is done it will be removed (hurrah). I've added a comment to this effect.